pyccel/ast/core.py

```diff
@@ -337,6 +346,7 @@ def __init__(self, *args):
 
 class PyccelOperator(Expr, PyccelAstNode):
 
+    __slots__ = ['_args', '_dtype', '_rank', '_shape'] 
```

Comment:
`PyccelAstNode` will also need a `__slots__` attribute for this to serve a purpose